src/mlpack/core/optimizers/cmaes/cmaes_impl.hpp

```diff
@@ -0,0 +1,922 @@
+      {
+         Log::Info << "TolUpX: standard deviation increased by more than "
+            << stopTolUpXFactor << ", larger initial standard"
+            << "deviation recommended."
+            << std::endl;
+            return true;
+        break;
+      }
+    }
+
+    // Condition of C greater than dMaxSignifKond
+    if (maxEW >= minEW* dMaxSignifKond)
+    {
+       Log::Info << "ConditionNumber: maximal condition number " <<
+       dMaxSignifKond << " reached. maxEW=" << maxEW <<  ",minEW="
+       << minEW << ",maxdiagC=" << maxdiagC << ",mindiagC="
+       << mindiagC << std::endl;
+        return true;
+    }
+
+    // Principal axis i has no effect on xmean
+    // ie. x == x + 0.1* sigma* rgD[i]* B[i]
+    if (!diag)
+    {
+      for (iAchse = 0; iAchse < N; ++iAchse)
+      {
+        fac = 0.1* sigma* rgD[iAchse];
+        for (iKoo = 0; iKoo < N; ++iKoo)
+        {
+          if (xmean[iKoo] != xmean[iKoo] + fac* B(iKoo, iAchse))
+            break;
+        }
+        if (iKoo == N)
+        {
+           Log::Info << "NoEffectAxis: standard deviation 0.1*" << (fac / 0.1)
+           << " in principal axis " << iAchse << " without effect"
+           << std::endl;
+           return true;
+          break;
+        }
+      }
+    }
+    // Component of xmean is not changed anymore
+    for (iKoo = 0; iKoo < N; ++iKoo)
+    {
+      if (xmean[iKoo] == xmean[iKoo] + sigma*
+        std::sqrt(C(iKoo, iKoo))/double(5))
+      {
+         Log::Info << "NoEffectCoordinate: standard deviation 0.2*"
+            << (sigma*std::sqrt(C(iKoo , iKoo))) << " in coordinate " << iKoo
+            << " without effect" << std::endl;
+        return true;
+        break;
+      }
+    }
+
+    if (countevals >= stopMaxFunEvals)
+    {
+       Log::Info << "MaxFunEvals: conducted function evaluations " << countevals
+          << " >= " << stopMaxFunEvals << std::endl;
+       return true;
+    }
+    if (gen >= stopMaxIter)
+    {
+       Log::Info << "MaxIter: number of iterations " << gen << " >= "
+          << stopMaxIter << std::endl;
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Conducts the eigendecomposition of C into B and D such that
+   * \f$C = B \cdot D \cdot D \cdot B^T\f$ and \f$B \cdot B^T = I\f$
+   * and D diagonal and positive.
+   * @param force For force == true the eigendecomposion is conducted even if
+   *              eigenvector and values seem to be up to date.
+   */
+void CMAES::updateEigensystem(bool force)
+  {
+    update();
+
+    if (!force)
+    {
+      if (eigensysIsUptodate)
+        return;
+      // return on modulo generation number
+      if (gen < genOfEigensysUpdate + updateCmode.modulo)
+        return;
+      // return on time percentage
+      if (updateCmode.maxtime < 1.00
+          && tictoctime > updateCmode.maxtime* totaltime
+          && tictoctime > 0.0002)
+        {
+          Log::Info << " time return happened " << std::endl;
```

Comment:
i wonder if that is necessary, since running more iterations shouldn't return false results. But I like the idea since we can save some time, maybe instead of using time as conditions we could use specify the number of iterations. What do you think?